helix-core/src/main/java/org/apache/helix/HelixCloudProperty.java

```diff
@@ -185,10 +213,20 @@ public void setCloudInfoSources(List<String> sources) {
     _cloudInfoSources = sources;
   }
 
+  @Deprecated
   public void setCloudInfoProcessorName(String cloudInfoProcessorName) {
     _cloudInfoProcessorName = cloudInfoProcessorName;
   }
 
+  /**
+   * Set the fully qualified class name of the cloud info processor.
+   * @param cloudInfoProcessorFullyQualifiedClassName
+   */
+  public void setCloudInfoProcessorFullyQualifiedClassName(
```

Comment:
Be careful with this API. Once you public, it may allow user set this dirtectly, if they have control on HelixCloudProperty. Since our public entries are only pkg and processor, let's not expose this set API but only get API.

Reply:
Made this private instead.